libnavigation-core/src/main/java/com/mapbox/navigation/directions/session/MapboxDirectionsSession.kt

```diff
@@ -65,6 +65,8 @@ class MapboxDirectionsSession(
             override fun onResponse(routes: List<DirectionsRoute>) {
                 currentRoutes = routes
                 state = State.RoutesAvailable
+                // start the timer only when the route has been requested at least once
+                fasterRouteTimer.start()
```

Comment:
Removing the timer from here and starting it only once is a change that I explicitly made. Are we not going to run into https://github.com/mapbox/driver/pull/374#discussion_r369471615?

Reply:
This is not going to start a separate countdown coroutine everytime. If you look at the `MapboxTimer` the `start` calls check if a timer is active and returns. Already tested the code and it works fine. I don't agree that we should start the fast route timer as soon as we instantiate `DirectionsSession`. The end user may never end up requesting for a route at all. In that case we will have a timer that will try to fetch route by itself.

Reply:
Gotcha :+1: